### ISSUE TITLE
User flow: write predicate operators in a TLA+ spec for which many traces can be generated 

### ISSUE THREAD

**Issue by danwt** [informalsystems/modelator-py#14]
This is a rephrasing of the 'traces/tests/counterexample' etc functionalities.

In https://github.com/informalsystems/modelator/commit/a351c2fe750216cd00867e3b63797fa9ed86b72e the user flow for writing 'tests' is as follows:

Write a spec something like this
```
------------------------------ MODULE IndicesHistoryTests ------------------------------
EXTENDS IndicesBalancesExec

VARIABLES 
  \* @type: Int;
  nsteps,
  \* @typeAlias: HISTORY = Int -> [action: ACTION, actionOutcome: Str, balances: BALANCES];
  \* @type: HISTORY;
  history


\* This predicate extends the Init predicate with history tracking
InitForTest ==
  /\ Init
  /\ nsteps = 0
  /\ history = [ n \in {0} |->
     [ action |-> action, actionOutcome |-> actionOutcome, balances |-> balances ]]

\* This predicate extends the Next predicate with history tracking
NextForTest ==
  /\ Next
  /\ nsteps' = nsteps + 1
  /\ history' = [ n \in DOMAIN history \union {nsteps'} |->
       IF n = nsteps' THEN
         [ action |-> action', actionOutcome |-> actionOutcome', balances |-> balances']
       ELSE history[n]
     ]

\* The view for the tests
\* @type: <<Str, Str>>;
View == << action.type, actionOutcome >>

\* The set of all actions in the history with the given type
Actions(type) ==
  { history[i]: i \in { i \in DOMAIN(history): history[i].action.type = type } }
    

Test2Steps ==
  nsteps = 2

Test3Steps ==
  nsteps = 3

Test2Claim ==
  Cardinality(Actions("Claim")) = 2

Test3Claim ==
  Cardinality(Actions("Claim")) = 3

Test2Free ==
  Cardinality(Actions("Free")) = 2

Test3Free ==
  Cardinality(Actions("Free")) = 3

Test2Freeze ==
  Cardinality(Actions("Freeze")) = 2

Test3Freeze ==
  Cardinality(Actions("Freeze")) = 3

TestIndexTakenWithoutReserveAndForceFreeze ==
  /\ \E index \in TakenIndices:
       balances[indices[index].who].reserved = 0
  /\ Actions("ForceTransfer") = {}
  /\ Actions("Freeze") = {}


TestTransferFundsViaIndices ==
  \E a1, a2 \in Addresses:
  \E s1, s2 \in DOMAIN history:
  LET S1 == history[s1] IN 
  LET S2 == history[s2] IN
    /\ S1.balances[a1].free + S1.balances[a2].free = S2.balances[a1].free + S2.balances[a2].free
    /\ S1.balances[a1].free /= S2.balances[a1].free
    /\ \A a \in Addresses:
         /\ S1.balances[a].reserved = S2.balances[a].reserved
         /\ a = a1 \/ a = a2 \/ S1.balances[a].free = S2.balances[a].free


=============================================================================
```

Calling `modelator trace IndicesHistoryTests.tla <cfg>` is a complex operation. It
1. Does some adhoc parsing to find operators prefixed or suffixed with 'Test'
2. Creates a .tla and cfg files that extend the original spec, and add a logically negated operator
3. The new .tla and .cfg files are checked
4. Counterexamples are parsed and returned.

I propose that step 1 and the overall user flow can be changed in the following ways.

1. We no longer use the terminology 'test'
2. A user can generate trace(s) that are executions whose final state satisfies a predicate P by writing an operator with _any_ name in any of the .tla files that are included in the spec (via extends for example). But they must prefix the operator name with a `\*` comment and a keyword. Like `\* @trace` (ideas welcome).

Put simply I think that moving forward users should specify _trace_ operators written like
```
------------------------------ MODULE foobar --------------------------------

EXTENDS bizz, bozz

\* arbitrary TLA+..

\* @trace
CrashedChain == ...

\* @trace
NegativeFunds == ...

===============================================================================
```

Moreover I think we should reserve the 'test' terminology for actual tests running in the repo of the SUT
